src/Symfony/Component/HttpKernel/EventListener/ErrorListener.php

```diff
@@ -32,19 +32,30 @@ class ErrorListener implements EventSubscriberInterface
     protected $controller;
     protected $logger;
     protected $debug;
+    protected $exceptionsMapping;
 
-    public function __construct($controller, LoggerInterface $logger = null, bool $debug = false)
+    public function __construct($controller, LoggerInterface $logger = null, bool $debug = false, array $exceptionsMapping = [])
     {
         $this->controller = $controller;
         $this->logger = $logger;
         $this->debug = $debug;
+        $this->exceptionsMapping = $exceptionsMapping;
     }
 
     public function logKernelException(ExceptionEvent $event)
     {
-        $e = FlattenException::createFromThrowable($event->getThrowable());
+        $throwable = $event->getThrowable();
+        $logLevel = null;
+        foreach ($this->exceptionsMapping  as $exception => $config) {
```

Comment:
Double space

```suggestion
        foreach ($this->exceptionsMapping as $exception => $config) {
```

Reply:
Maybe even rename `$exception` to `$class`... But that is just a matter of taste.

I think it would be more readable that way.